### PR TITLE
Adding curve checks to integratedTests

### PR DESCRIPTION
**Pull request by cssherman**
integratedTests PR: https://github.com/GEOS-DEV/integratedTests/pull/7

### CODE REVIEW

File: inputFiles/solidMechanics/beamBending_curve.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+def convert_bulk_shear(bulk_modulus, shear_modulus):
+    return 9.0 * bulk_modulus * shear_modulus / (3.0 * bulk_modulus + shear_modulus)
+
+
+def max_bending(x, youngs_modulus, moment_inertia, beam_length, force):
+    return (force * x * x) * (3.0 * beam_length - x) / (6.0 * youngs_modulus * moment_inertia)
+
+
+def curve(**kwargs):
```
Comment (TotoGaz) [inline inputFiles/solidMechanics/beamBending_curve.py:13]:
```suggestion
def create_curve(**kwargs):
```
?

**Review by TotoGaz — APPROVED** [review]

**Comment by cssherman** [comment]
@rrsettgast @TotoGaz - here is an example of the latest output for curvechecks with a baseline and a script-generated curve.  I'm using one of the beam bending problems as an example, with the following ats input:

```python
curvecheck_params = {}
curvecheck_params['filename'] = 'displacement_history.hdf5'
curvecheck_params['tolerance'] = 1e-10
curvecheck_params['script_instructions'] = [['./beamBending_curve.py', 'curve', 'totalDisplacement', 'trace']]
curvecheck_params['script_tolerance'] = 0.001
curvecheck_params['curves'] = [['totalDisplacement', 'trace']]

...

TestCase(name="beamBending_%02d" % N,
             desc="Tests beam bending on %d ranks." % N,
             label="auto",
             owner="Ben Corbett",
             independent=True,
             steps=(geos(deck="beamBending.xml",
                         name="beamBending",
                         np=N,
                         ngpu=N,
                         x_partitions=nx,
                         y_partitions=ny,
                         z_partitions=nz,
                         restartcheck_params=restartcheck_params,
                         curvecheck_params=curvecheck_params), ))
```

Since this problem has both a baseline and a script-generated curve, there are three types of output (note the very small axis for the 0th component):

image

For the image generation, I check the number of points and timesteps we have to plot, and use the larger of the two to set the x-axis (currently Time or X) and use the smaller to generate the different colored curves.  Looking at the above example, I want to add a colorbar to indicate this second scale.

**Comment by cssherman** [comment]
@CusiniM - these new curve checks are in a decent state now.  I've also tried my hand at updating a couple of the hydrofracture tests to work with the curvechecks and the existing analytic solutions.  Could you give this branch a shot and see how you feel about the new input format and results?

FYI - the baseline restartcheck files are a bit behind develop, which causes many of the checks to fail.  I'll work on updating them once you and @rrsettgast are happy.

**Comment by cssherman** [comment]
> @CusiniM - these new curve checks are in a decent state now. I've also tried my hand at updating a couple of the hydrofracture tests to work with the curvechecks and the existing analytic solutions. Could you give this branch a shot and see how you feel about the new input format and results?
> 
> FYI - the baseline restartcheck files are a bit behind develop, which causes many of the checks to fail. I'll work on updating them once you and @rrsettgast are happy.

Also, some of the hydrofracture examples still don't compare well to the analytic solutions (see the PKN example).  If you have any thoughts on changes to those tests, I'd be happy to hear them.

**Comment by CusiniM** [comment]
> > @CusiniM - these new curve checks are in a decent state now. I've also tried my hand at updating a couple of the hydrofracture tests to work with the curvechecks and the existing analytic solutions. Could you give this branch a shot and see how you feel about the new input format and results?
> > FYI - the baseline restartcheck files are a bit behind develop, which causes many of the checks to fail. I'll work on updating them once you and @rrsettgast are happy.
> 
> Also, some of the hydrofracture examples still don't compare well to the analytic solutions (see the PKN example). If you have any thoughts on changes to those tests, I'd be happy to hear them.

Is it maybe just a resolution issue? I don't really expect 


> > @CusiniM - these new curve checks are in a decent state now. I've also tried my hand at updating a couple of the hydrofracture tests to work with the curvechecks and the existing analytic solutions. Could you give this branch a shot and see how you feel about the new input format and results?
> > FYI - the baseline restartcheck files are a bit behind develop, which causes many of the checks to fail. I'll work on updating them once you and @rrsettgast are happy.
> 
> Also, some of the hydrofracture examples still don't compare well to the analytic solutions (see the PKN example). If you have any thoughts on changes to those tests, I'd be happy to hear them.

Could it just be a resolution issue? I don't really expect the meshes used in those smoke tests to give very good matches.

**Comment by CusiniM** [comment]
> @rrsettgast @TotoGaz - here is an example of the latest output for curvechecks with a baseline and a script-generated curve. I'm using one of the beam bending problems as an example, with the following ats input:
> 
> ```python
> curvecheck_params = {}
> curvecheck_params['filename'] = 'displacement_history.hdf5'
> curvecheck_params['tolerance'] = 1e-10
> curvecheck_params['script_instructions'] = [['./beamBending_curve.py', 'curve', 'totalDisplacement', 'trace']]
> curvecheck_params['script_tolerance'] = 0.001
> curvecheck_params['curves'] = [['totalDisplacement', 'trace']]
> 
> ...
> 
> TestCase(name="beamBending_%02d" % N,
>              desc="Tests beam bending on %d ranks." % N,
>              label="auto",
>              owner="Ben Corbett",
>              independent=True,
>              steps=(geos(deck="beamBending.xml",
>                          name="beamBending",
>                          np=N,
>                          ngpu=N,
>                          x_partitions=nx,
>                          y_partitions=ny,
>                          z_partitions=nz,
>                          restartcheck_params=restartcheck_params,
>                          curvecheck_params=curvecheck_params), ))
> ```
> 
> Since this problem has both a baseline and a script-generated curve, there are three types of output (note the very small axis for the 0th component):
> 
> image
> 
> For the image generation, I check the number of points and timesteps we have to plot, and use the larger of the two to set the x-axis (currently Time or X) and use the smaller to generate the different colored curves. Looking at the above example, I want to add a colorbar to indicate this second scale.

Maybe instead of script we could use `analytical` or `reference`?

**Review by CusiniM — APPROVED** [review]

**Pull request merged**